### ISSUE TITLE
Allow plugins to bail out of custom message and do it themselfs with substitutions

### ISSUE THREAD

**Issue by AndyOGo** [stylelint/stylelint#5170]
> What is the problem you're trying to solve?

My plugin provides a custom `message` with substituted interpolations.
Unfortunately they are overruled by [stylelint's `message` option](https://stylelint.io/user-guide/configure#message) introduced with https://github.com/stylelint/stylelint/pull/672

Support for substitutions has never been added and the request for it got closed #836

Seem that it is ignored because of this [general solution](https://github.com/stylelint/stylelint/pull/672/files#diff-78f1c80ffb2836008dd194b3b0ca28f9b46e4897b606f0b3d25a29e57a8d3e61R74), so I deleted the `customMessages` entry for my rule, like:

```js
delete result.stylelint.customMessages[ruleName];
```

> What solution would you like to see?

Well, some official way of saying, hey stylelint this plugin is handling the `message` option by itself.

Original issue in my plugin: https://github.com/AndyOGo/stylelint-declaration-strict-value/issues/142

**Comment by jeddy3** [comment]
@AndyOGo Thanks for the request.

Labelling as a discussion for now. Ideally, we would provide a common way for using interpolated values in custom messages that all plugins can use, but in the meantime, an official flag may by the best option.

> Support for substitutions has never been added and the request for it got closed #836

It picked up again in https://github.com/stylelint/stylelint/issues/4117, but the issue has stalled.

**Comment by SalimBensiali** [comment]
I would love the ability to supplement the default message with extra info and recommendation. We have RTL aware Sass variables that we want developers to use instead of `margin-left`, `border-left-color` etc. And a custom message currently lacks the ability to preserve the existing message. Something with string interpolation or an `additional_message` option would be grand.

I would love to be able to do the following for example
```json
{
  "rules": {
    "property-disallowed-list": [
      "/left|right$/",
      {
        "message": "${original_message}. Use the \"$start-direction\" & \"$end-direction\" SASS variables instead of \"left\" & \"right\".",
      }
    ]
  }
}
```

or 

```json
{
  "rules": {
    "property-disallowed-list": [
      "/left|right$/",
      {
        "additional_message": "Use the \"$start-direction\" & \"$end-direction\" SASS variables instead of \"left\" & \"right\".",
      }
    ]
  }
}
```

**Comment by SalimBensiali** [comment]
@jeddy3 Is there anything I can help with on this one? I would be keen to contribute, or is this needing internal discussion first?

Thanks.

**Comment by jeddy3** [comment]
@SalimBensiali Thanks for offering to contribute. The original interpolated custom messages issue (https://github.com/stylelint/stylelint/issues/4117) stalled, but a proof-of-concept pull request showing how this might work would be a good start in getting it going again. It is probably worth seeing if this problem has been solved in other tools, and applying their approach here in stylelint.

**Comment by SalimBensiali** [comment]
Thanks @jeddy3. I will do some research and raise a POC PR afterwards.

**Comment by Mouvedia** [comment]
@ybiquitous is this achievable since #6312 merge?

**Comment by ybiquitous** [comment]
Yes, let's close this. The functional `message` support should resolve this issue.

https://github.com/stylelint/stylelint/blob/68cb920c45641702d76670afb111960cbd743296/docs/user-guide/configure.md?plain=1#L150-L160
